### PR TITLE
feat: deno install in Rust

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: cli/installer.rs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  file_path: PathBuf,
+  commands: Vec<String>,
+) -> Result<(), Error> {
+  // On Windows if user is using Powershell .cmd extension is need to run the
+  // installed module.
+  // Generate batch script to satisfy that.
+  let template_header = "This executable is generated by Deno. Please don't modify it unless you know what it means.";
+
+  if cfg!(windows) {
+    let template = format!(
```
Comment (ry) [inline cli/installer.rs]: Add TODO remove the conditional in generated script. Script should be simple.

File: cli/installer.rs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      r#"#!/bin/sh
+# {}
+basedir=$(dirname "$(echo "$0" | sed -e 's,\\\\,/,g')")
+
+case \`uname\` in
+  *CYGWIN*) basedir=\`cygpath -w "$basedir"\`;;
+esac
+
+if [ -x "$basedir/deno" ]; then
```
Comment (ry) [inline cli/installer.rs]: Add TODO remove the conditional in generated script. Script should be simple.

File: std/installer/test.ts [hunk 3]
```diff
@@ -1,472 +0,0 @@
...
-    const s = new TextDecoder("utf-8").decode(b);
-
-    assertEquals(s, "hello, foo");
-  } catch (err) {
-    console.error(err);
-    thrown = true;
-  } finally {
-    await remove(tempDir, { recursive: true });
-    ps.close();
-  }
```
Comment (bartlomieju) [inline std/installer/test.ts]: @ry not sure how to test it, those tests depend on `deno` being in `PATH`...

Comment (ry) [reply std/installer/test.ts]: Can you add target/debug to the PATH? (`current_executable().parent().unwrap()`)

Comment (bartlomieju) [reply std/installer/test.ts]: That's doable, but tests are run in parallel, I'm not sure messing with env variables is a good idea :/

Comment (ry) [reply std/installer/test.ts]: So the test is to actually call the shell script? You'll have to do this with a custom integration test (non-itest). When you spawn the subprocess to execute the shell script, you'll be able to specify the env.

Comment (bartlomieju) [reply std/installer/test.ts]: Obiviously... thanks 👍 still not sure what causes panic in current tests

**Pull request merged**